packages/core/admin/utils/should-build-admin.js

```diff
@@ -31,11 +32,11 @@ const hasNonDefaultPlugins = plugins => {
 
 const hasCustomAdminCode = async (dir, useTypeScript) => {
   const customAdminPath = path.join(dir, 'src', 'admin');
-  const customAdminConfigFileExtension = useTypeScript ? 'app.tsx' : 'app.js';
-  const customAdminConfigFile = path.join(customAdminPath, customAdminConfigFileExtension);
+
+  const customAdminAppConfigFile = await getCustomAppConfigFile(dir, useTypeScript);
   const customAdminWebpackFile = path.join(customAdminPath, 'webpack.config.js');
 
-  const hasCustomConfigFile = await fs.pathExists(customAdminConfigFile);
+  const hasCustomConfigFile = customAdminAppConfigFile !== undefined;
```

Comment:
I think this could be reduced to:

```suggestion
  const hasCustomConfigFile = !!customAdminAppConfigFile;
```